python/mlc_llm/serve/entrypoints/entrypoint_utils.py

```diff
@@ -86,7 +94,39 @@ def process_prompts(
         is_token_ids = isinstance(input_prompt, list) and all(
             isinstance(token_id, int) for token_id in input_prompt
         )
-        if not (is_str or is_token_ids):
+        is_image = isinstance(input_prompt, data.ImageData)
+        if not (is_str or is_token_ids or is_image):
             return create_error_response(HTTPStatus.BAD_REQUEST, message=error_msg)
         output_prompts.append(ftokenize(input_prompt) if is_str else input_prompt)  # type: ignore
     return output_prompts
+
+
+def get_image_from_url(url: str):
+    """Get the image from the given URL, process and return the image tensor as TVM NDArray."""
+
+    # pylint: disable=import-outside-toplevel, import-error
+    from PIL import Image
+    from transformers import CLIPImageProcessor
+
+    response = requests.get(url, timeout=5)
+    image_tensor = Image.open(BytesIO(response.content)).convert("RGB")
+
+    image_processor = CLIPImageProcessor(
+        size={"shortest_edge": 336}, crop_size={"height": 336, "width": 336}
+    )
+    image_features = tvm.nd.array(
+        image_processor.preprocess(image_tensor, return_tensors="np")["pixel_values"].astype(
+            "float16"
+        )
+    )
+    return image_features
+
+
+def get_image_embed_size(config_file_path: str) -> int:
+    """Get the image embedding size from the model config file."""
+    with open(config_file_path, "r", encoding="utf-8") as file:
+        config = json.load(file)
+    image_size = config["model_config"]["vision_config"]["image_size"]
+    patch_size = config["model_config"]["vision_config"]["patch_size"]
+    embed_size = (image_size // patch_size) ** 2
+    return embed_size
```

Comment:
Check out the other comment. Here we can accept the config dict and reduce the need of loading JSON.